homeassistant/components/point/config_flow.py

```diff
@@ -0,0 +1,197 @@
+
+        self.flow_impl = DOMAIN
+
+        return await self.async_step_auth()
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        flows = self.hass.data.get(DATA_FLOW_IMPL, {})
+
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='already_setup')
+
+        if not flows:
+            _LOGGER.debug("no flows")
+            return self.async_abort(reason='no_flows')
+
+        if len(flows) == 1:
+            self.flow_impl = list(flows)[0]
+            return await self.async_step_auth()
+
+        if user_input is not None:
+            self.flow_impl = user_input['flow_impl']
+            return await self.async_step_auth()
+
+        return self.async_show_form(
+            step_id='user',
+            data_schema=vol.Schema({
+                vol.Required('flow_impl'):
+                vol.In(list(flows))
+            }))
+
+    async def async_step_auth(self, user_input=None):
+        """Create an entry for auth."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='external_setup')
+
+        errors = {}
+
+        if user_input is not None:
+            errors['base'] = 'follow_link'
+
+        try:
+            with async_timeout.timeout(10):
+                url = await self._get_authorization_url()
+        except asyncio.TimeoutError:
+            return self.async_abort(reason='authorize_url_timeout')
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected error generating auth url")
+            return self.async_abort(reason='authorize_url_fail')
+
+        return self.async_show_form(
+            step_id='auth',
+            description_placeholders={'authorization_url': url},
+            errors=errors,
+        )
+
+    async def _get_authorization_url(self):
+        """Create Minut Point session and get authorization url."""
+        from pypoint import PointSession
+        flow = self.hass.data[DATA_FLOW_IMPL][self.flow_impl]
+        client_id = flow[CLIENT_ID]
+        client_secret = flow[CLIENT_SECRET]
+        point_session = PointSession(
+            client_id, client_secret=client_secret)
+
+        self.hass.http.register_view(MinutAuthCallbackView())
+
+        return point_session.get_authorization_url
+
+    async def async_step_code(self, code=None):
+        """Received code for authentication."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='already_setup')
+
+        if code is None:
+            return self.async_abort(reason='no_code')
+
+        _LOGGER.debug("Should close all flows below %s",
+                      self.hass.config_entries.flow.async_progress())
+        # Remove notification if no other discovery config entries in progress
+
+        return self._create_session(code)
```

Comment:
Any reason you extracted this into an extra method ?

Reply:
I thought it would be used by another method, but it turned out not to be the case..

Do you want me to merge it with the method above?

Reply:
Ah, I didn't want to do:
 `
from pypoint import PointSession` 
in 
`async def async_step_code(self, code=None):` 
unless all parts were successful (but perhaps that doesn't matter).